cdap-ui/app/cdap/components/SpotlightSearch/SpotlightModal/index.js

```diff
@@ -0,0 +1,255 @@
+/*
+ * Copyright © 2016 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+import React, {Component, PropTypes} from 'react';
+import {MySearchApi} from 'api/search';
+import {default as NamespaceStore} from 'services/store/store';
+import {parseMetadata} from 'services/metadata-parser';
+import T from 'i18n-react';
+import Mousetrap from 'mousetrap';
+import classnames from 'classnames';
+import {
+  Col,
+  Modal,
+  ModalHeader,
+  ModalBody,
+  Tag,
+  Pagination,
+  PaginationItem,
+  PaginationLink
+} from 'reactstrap';
+
+require('./SpotlightModal.less');
+
+const PAGE_SIZE = 10;
+
+export default class SpotlightModal extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      searchResults: { results: [] },
+      currentPage: 1,
+      numPages: 1,
+      focusIndex: 0
+    };
+  }
+
+  componentWillMount() {
+    Mousetrap.bind('right', this.handleNext.bind(this));
+    Mousetrap.bind('left', this.handlePrev.bind(this));
+    Mousetrap.bind('up', this.handleUpDownArrow.bind(this, 'UP'));
+    Mousetrap.bind('down', this.handleUpDownArrow.bind(this, 'DOWN'));
+    Mousetrap.bind('enter', this.handleEnter.bind(this));
+
+    this.handleSearch(1);
+  }
+
+  componentWillUnmount() {
+    Mousetrap.unbind('right');
+    Mousetrap.unbind('left');
+    Mousetrap.unbind('up');
+    Mousetrap.unbind('down');
+    Mousetrap.unbind('enter');
+  }
+
+  handleNext() {
+    this.handleSearch(this.state.currentPage + 1);
+  }
+
+  handlePrev() {
+    this.handleSearch(this.state.currentPage - 1);
+  }
+
+  handleUpDownArrow(direction) {
+    if (direction === 'UP') {
+      let focusIndex = this.state.focusIndex === 0 ? 0 : this.state.focusIndex - 1;
+      this.setState({focusIndex});
+    } else if (direction ==='DOWN') {
+      let totalResults = this.state.searchResults.results.length;
+
+      let focusIndex = this.state.focusIndex === totalResults - 1 ?
+        this.state.focusIndex : this.state.focusIndex + 1;
+
+      this.setState({focusIndex});
+    }
+  }
+
+  handleEnter() {
+    let entity = parseMetadata(this.state.searchResults.results[this.state.focusIndex]);
+
+    // Redirect to entity once we have entity detail page
+    // Right now console logging entity for identification purposes
+
+    console.log('ENTER on:', entity.id);
```

Comment:
This is for identifier purpose, so that we know the function works since we can't do the real implementation yet